code/parse/sexp.cpp

```diff
@@ -14486,18 +14531,17 @@ bool sexp_check_flag_arrays(char *flag_name, Object::Object_Flags &object_flag,
 
 int sexp_are_ship_flags_set(int node)
 {
-	char *flag_name;
 	Object::Object_Flags object_flag = Object::Object_Flags::NUM_VALUES;
     Ship::Ship_Flags ship_flags = Ship::Ship_Flags::NUM_VALUES;
 	Mission::Parse_Object_Flags parse_obj_flag = Mission::Parse_Object_Flags::NUM_VALUES;
 	AI::AI_Flags ai_flag = AI::AI_Flags::NUM_VALUES;
 
 	auto ship_entry = eval_ship(node);
 	if (!ship_entry || ship_entry->status == ShipStatus::EXITED) {
-		return SEXP_KNOWN_FALSE;
+		return SEXP_NAN_FOREVER;
 	}
 	if (ship_entry->status == ShipStatus::NOT_YET_PRESENT) {
-		return SEXP_CANT_EVAL;
+		return SEXP_NAN;
```

Comment:
Are these intentional? Seems odd to return NaN from a SEXP that returns a boolean value.

Reply:
It's intentional.  In fact I had to make this change in a few places in the SEXP refactor (including commits that aren't yet in PRs).  I checked the documentation and I also went back to the original public code release to make sure I had the right idea.

SEXP_NAN represents an invalid value, or no value, because it's not possible to determine what the number is supposed to be, or what the boolean value is supposed to be.  SEXP_CANT_EVAL means that the event system defers evaluating the event until some future time.  CANT_EVAL is used to prevent directives from immediately appearing, and it is almost exclusively used for "objective" type SEXPs like `has-destroyed-delay`.

Reply:
Thanks for the explanaition.